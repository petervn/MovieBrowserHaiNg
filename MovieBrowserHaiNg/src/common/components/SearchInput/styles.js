import {StyleSheet} from 'react-native';

import {darkBlue, freeze} from '../../styles/color';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 5,
    paddingBottom: 5,
  },
  containerInput: {
    height: 40,
    backgroundColor: freeze,
    borderRadius: 15,
  },
  inputDirection: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginLeft: 10,
    marginRight: 10,
  },
  icon: {
    padding: 10,
  },
  textInput: {
    flex: 1,
    height: '100%',
    fontSize: 12,
    color: darkBlue,
    width: '100%',
  },
});

export default styles;
