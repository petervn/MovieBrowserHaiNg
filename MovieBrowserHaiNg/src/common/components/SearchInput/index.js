import React, {useState} from 'react';
import {View, TextInput, Image, Text, TouchableOpacity} from 'react-native';
import SearchIcon from '../../../common/asset/images/search-icon.png';

import {darkGray} from '../../styles/color';
import styles from './styles';

// todo. It is better If We use search bar component from 'react-native-elements'
// https://react-native-training.github.io/react-native-elements/docs/searchbar.html
export default SearchInput = ({onSubmit, onSeachTextChange}) => {
  const [value, setValue] = useState('');

  const onChangeText = textValue => {
    setValue(textValue);
    onSeachTextChange(textValue);
  };

  return (
    <View style={styles.container}>
      <View style={styles.containerInput}>
        <View style={styles.inputDirection}>
          <Image
            source={SearchIcon}
            style={{width: 20, height: 20, marginRight: 10}}
            sizeMode="contain"
          />
          <TextInput
            style={styles.textInput}
            onSubmitEditing={() => {}}
            onChangeText={textValue => onChangeText(textValue)}
            value={value}
            returnKeyType="search"
            keyboardType="default"
            multiline={false}
            autoCorrect={false}
            placeholderTextColor={darkGray}
            placeholder="Search"
          />
          {value.length > 0 && (
            <TouchableOpacity onPress={() => setValue('')}>
              <Text>X</Text>
            </TouchableOpacity>
          )}
        </View>
      </View>
    </View>
  );
};
