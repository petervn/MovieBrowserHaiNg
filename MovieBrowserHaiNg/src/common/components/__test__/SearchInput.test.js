import React from 'react';
import {TouchableOpacity, TextInput} from 'react-native';
import {shallow} from 'enzyme';
import {expect as chaiExpect} from 'chai';
import SearchInput from '../SearchInput';

describe('SearchInput', () => {
  const onSubmit = jest.fn(),
    onSeachTextChange = jest.fn();

  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should trigger onSeachTextChange correctly', () => {
      const component = mount(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      component
        .find(SearchInput)
        .props()
        .onSeachTextChange('search value');
      expect(onSeachTextChange).toHaveBeenCalled();

      component
        .find(TextInput)
        .props()
        .onChangeText('input value');

      expect(onSeachTextChange).toHaveBeenCalled();
    });

    it('should hide clear X icon when value is empty', () => {
      let component = shallow(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      chaiExpect(component.find(TouchableOpacity)).to.have.lengthOf(0);
    });

    it('should show clear X icon when value is not empty', () => {
      let component = shallow(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      component
        .find(TextInput)
        .props()
        .onChangeText('hi');
      chaiExpect(component.find(TouchableOpacity)).to.have.lengthOf(1);
    });

    it('should hide clear X icon when value back to be empty', () => {
      let component = shallow(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      component
        .find(TextInput)
        .props()
        .onChangeText('hi');
      chaiExpect(component.find(TouchableOpacity)).to.have.lengthOf(1);
      component
        .find(TextInput)
        .props()
        .onChangeText('');
      chaiExpect(component.find(TouchableOpacity)).to.have.lengthOf(0);
    });

    it('should show value when user key in', () => {
      let component = shallow(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      component
        .find(TextInput)
        .props()
        .onChangeText('hi');
      expect(component.find(TextInput).props().value).toEqual('hi');
    });
    it('should show placeholder', () => {
      let component = shallow(
        <SearchInput
          onSubmit={onSubmit}
          onSeachTextChange={onSeachTextChange}
        />,
      );
      expect(component.find(TextInput).props().placeholder).toEqual('Search');
    });
  });
});
