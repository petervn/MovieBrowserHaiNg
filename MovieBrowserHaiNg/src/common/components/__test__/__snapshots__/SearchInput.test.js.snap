// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "paddingBottom": 5,
      "paddingHorizontal": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
        "borderRadius": 15,
        "height": 40,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <Image
        sizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/common/asset/images/search-icon.png",
          }
        }
        style={
          Object {
            "height": 20,
            "marginRight": 10,
            "width": 20,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        keyboardType="default"
        multiline={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Search"
        placeholderTextColor="#a1a1a4"
        rejectResponderTermination={true}
        returnKeyType="search"
        style={
          Object {
            "color": "#47525e",
            "flex": 1,
            "fontSize": 12,
            "height": "100%",
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;
